packages/medusa/src/services/product-category.ts

```diff
@@ -217,6 +217,27 @@ class ProductCategoryService extends TransactionBaseService {
         })
     })
   }
+
+  /**
+   * Add a batch of product to a product category
+   * @param productCategoryId - The id of the product category on which to add the products
+   * @param productIds - The products ids to attach to the product category
+   * @return the product category on which the products have been added
+   */
+  async addProducts(
+    productCategoryId: string,
+    productIds: string[],
+    config: FindConfig<ProductCategory> = {},
+  ): Promise<ProductCategory | never> {
+    return await this.atomicPhase_(async (manager) => {
+      const productCategoryRepository: ProductCategoryRepository =
+        manager.getCustomRepository(this.productCategoryRepo_)
+
+      await productCategoryRepository.addProducts(productCategoryId, productIds)
+
+      return await this.retrieve(productCategoryId, config)
```

Comment:
suggestion: I think that this method should return void and then if the consumer wants to retrieve it he can call retrieve with the right arguments. It will avoid mixing mutation arguments and find arguments. plus it is possible that the consumer does not want to retrieve them immediately. Usually, we split that in two actions in the endpoint handler, the mutation occurs in the transaction and the retrieve outside the transaction and we pass the different config. Also, at the moment, the fields are accepted by the API but they are not used either.

Reply:
+1

Reply:
Void was my choice as well. This would fall into the same comment I made above - https://github.com/medusajs/medusa/pull/3123#discussion_r1088828262

Other similar cases do the same and I don't want to introduce a deviation. I'll add this to the same issue i'll be creating for the previous comment.

Reply:
we also currently changed some of those that are doing that to void, in some cases it make more sense and in other cases it does not